packages/eas-cli/src/metadata/utils/__tests__/telemetry.test.ts

```diff
@@ -1,23 +1,73 @@
-import { App } from '@expo/apple-utils';
+import { App, getRequestClient } from '@expo/apple-utils';
 
-import { TelemetryContext, makeDataScrubber } from '../telemetry';
+import { MetadataEvent } from '../../../analytics/events';
+import { MetadataTelemetryError } from '../../errors';
+import { TelemetryContext, makeDataScrubber, withTelemetryAsync } from '../telemetry';
 
-describe(makeDataScrubber, () => {
-  const stub: TelemetryContext = {
-    // Only the App ID is considered to be sensitive
-    app: new App({} as any, 'SECRET_APP_ID', {} as any),
-    // Only the token properties and user credentials are considered to be sensitive
-    auth: {
-      username: 'SECRET_USERNAME',
-      password: 'SECRET_PASSWORD',
-      context: {
-        token: 'SECRET_TOKEN',
-        teamId: 'SECRET_TEAM_ID',
-        providerId: 1337,
-      },
-    } as any,
-  };
+const stub: TelemetryContext = {
+  // Only the App ID is considered to be sensitive
+  app: new App({} as any, 'SECRET_APP_ID', {} as any),
+  // Only the token properties and user credentials are considered to be sensitive
+  auth: {
+    username: 'SECRET_USERNAME',
+    password: 'SECRET_PASSWORD',
+    context: {
+      token: 'SECRET_TOKEN',
+      teamId: 'SECRET_TEAM_ID',
+      providerId: 1337,
+    },
+  } as any,
+};
+
+describe(withTelemetryAsync, () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('detaches interceptors after action resolved', async () => {
+    const { interceptors } = getRequestClient();
+    const ejectInterceptor = jest.spyOn(interceptors.response, 'eject');
+
+    await withTelemetryAsync(MetadataEvent.APPLE_METADATA_DOWNLOAD, stub, async () => {
+      return 'testing resolved telemetry action';
+    });
+
+    expect(ejectInterceptor).toBeCalledWith(expect.any(Number));
+  });
+
+  it('detaches interceptors after action rejected', async () => {
+    const { interceptors } = getRequestClient();
+    const ejectInterceptor = jest.spyOn(interceptors.response, 'eject');
 
+    try {
+      await withTelemetryAsync(MetadataEvent.APPLE_METADATA_DOWNLOAD, stub, async () => {
+        throw new Error('testing rejected telemetry action');
+      });
+    } catch {
+      // intentional failure
+    }
+
+    expect(ejectInterceptor).toBeCalledWith(expect.any(Number));
+  });
+
+  it('adds executionId to thrown errors', async () => {
+    let caughtError: null | Error = null;
+
+    try {
+      await withTelemetryAsync(MetadataEvent.APPLE_METADATA_DOWNLOAD, stub, async () => {
+        throw new Error('testing exectution ID');
+      });
+    } catch (error: any) {
+      caughtError = error;
+    }
+
+    expect(caughtError).toBeInstanceOf(MetadataTelemetryError);
+    expect(caughtError).toHaveProperty('executionId', expect.any(String));
+    expect(caughtError).toHaveProperty('originalError', expect.any(Error));
```

Comment:
`await expect(...).rejects.toThrow(...)`

Reply:
I had this at first, but in we also need to test the "executionId". I don't think we can do such a thing with just that line, right?

Reply:
I see your point but you're just checking if `executionId` and `originalError` were set to anything. If we're using typescript for all files in this repo then it's not possible to create an instance of `MetadataTelemetryError` without specifying those two fields.

I'm not feeling too strong about this though.